SYCL/Basic/alloc_on_subdevice.cpp

```diff
@@ -0,0 +1,32 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env CreateMultipleSubDevices=2 NEOReadDebugKeys=1 SYCL_DEVICE_FILTER="gpu" %t.out
+
+// REQUIRES: gpu
+
+#include <CL/sycl.hpp>
+
+#include <iostream>
+using namespace sycl;
+
+class KernelNameA;
+int main() {
+  device dev = device::get_devices(info::device_type::gpu)[0];
+  int val = 0;
+  std::vector<device> subdevices;
+
+  subdevices = dev.create_sub_devices<
+      info::partition_property::partition_by_affinity_domain>(
+      info::partition_affinity_domain::numa);
+  std::cout << "Testing on " << subdevices.size() << " subdevices" << std::endl;
+
+  context commonContext{subdevices};
+  for (int j = 0; j < subdevices.size(); ++j) {
+    buffer<int, 1> buf(&val, range<1>(1));
+    queue q{commonContext, subdevices[j]};
+    q.submit([&](handler &cgh) {
+      auto acc = buf.get_access<access::mode::read_write>(cgh);
+      cgh.single_task<KernelNameA>([=]() { acc[0] = acc[0] + 1; });
+    });

```

Comment:
Got to add a wait call here.

Reply:
Is it strictly required? I think we will still wait at queue destruction, no?

Reply:
No, we won't wait on queue's destruction.
DPCPP RT waits on destruction of buffer. But I'm not sure if all use-cases involve this sort of wait.
@sergey-semenov , could you, please, help answering if we wait for buffer destruction in all cases?

Reply:
We do, subbuffers aside. No need for `queue::wait()` here since the SYCL spec requires us to wait for and copy the latest value to `val` on buffer destruction.